kratos/containers/container_expression/expressions/expression.h

```diff
@@ -37,14 +37,87 @@ class KRATOS_API(KRATOS_CORE) Expression {
     ///@name Type definitions
     ///@{
 
-    using Pointer = Kratos::intrusive_ptr<Expression>;
+    using Pointer = Kratos::intrusive_ptr<const Expression>;
 
     using IndexType = std::size_t;
 
+    ///@}
+    ///@name Public classes
+    ///@{
+
+    class ExpressionIterator {
+    public:
+        ///@name Type definitions
+        ///@{
+
+        KRATOS_CLASS_POINTER_DEFINITION(ExpressionIterator);
```

Comment:
I was too lazy to do all those.... :/ :D... Now I have to fix them :P 

Btw, all requirements cannot be satisfied, because, `std::forward_iterator` requires a default constructor as well which does not make sense in this case... :/

Reply:
You can add a default constructor that throws.

Reply:
done :)